src/signals/incident-management/containers/ReporterContainer/__tests__/useFetchReporter.test.ts

```diff
@@ -26,7 +30,69 @@ jest.mock('react-router-dom', () => ({
 
 const INCIDENT_ID = '4440'
 
+const REPORTER_MOCK: Result = {
+  id: 0,
+  created_at: new Date(0).toISOString(),
+  category: {
+    sub: 'foo',
+    sub_slug: 'foo',
+    main: 'foo',
+    main_slug: 'foo',
+    departments: 'foo',
+  },
+  status: {
+    state: 'foo',
+    state_display: 'foo',
+  },
+  feedback: null,
+  can_view_signal: false,
+  has_children: false,
+}
+
 describe('Fetch Reporter hook', () => {
+  it('correctly implements pagination', async () => {
+    server.use(
+      rest.get(
+        `${configuration.INCIDENT_PRIVATE_ENDPOINT}${INCIDENT_ID}/context/reporter`,
+        async (req, res, ctx) => {
+          const query = req.url.searchParams
+          const page = query.get('page')
+          return res(
+            ctx.status(200),
+            ctx.json({
+              count: 8,
+              results: [{ ...REPORTER_MOCK, id: Number(page) }],
+            })
+          )
+        }
+      )
+    )
+
+    const { result, waitForNextUpdate } = renderHook(
+      () => useFetchReporter(INCIDENT_ID),
+      {
+        wrapper: Provider,
+        initialProps: { store },
+      }
+    )
+
+    await waitForNextUpdate()
+    await waitForNextUpdate()
```

Comment:
Why do you need 2 cicli here?

Reply:
is there a way to prevent this? If I don't wait twice there are unhandled async events and it will complain about code not being wrapped in `act()` calls